pallets/parachain-staking/src/lib.rs

```diff
@@ -1444,6 +1443,68 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Notify a collator is inactive during MaxOfflineRounds
+		#[pallet::call_index(29)]
+		#[pallet::weight(
+			T::DbWeight::get().reads_writes(0 as u64, 0 as u64)
+		)]
+		pub fn notify_inactive_collator(
+			origin: OriginFor<T>,
+			collator: T::AccountId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let collators = <SelectedCandidates<T>>::get();
+			let max_collators = <TotalSelected<T>>::get();
+
+			// Check collators length is not below or eq to 66% of max_collators.
+			// We use saturating logic here with (2/3)
+			// as it is dangerous to use floating point numbers directly.
+			ensure!(
+				collators.len() * 3 > (max_collators * 2) as usize,
+				<Error<T>>::TooLowCollatorCountToNotifyAsInactive
+			);
+
+			let round_info = <Round<T>>::get();
+			let max_offline_rounds = T::MaxOfflineRounds::get();
+
+			// Take last round to have rounds_to_check = [7,8,9]
+			// in case we are in round 10 for instance
+			let round = round_info.current.saturating_sub(1);
+
+			let mut rounds_to_check: Vec<RoundIndex> = vec![];
+			for round_index in (0..max_offline_rounds).rev() {
+				rounds_to_check.push(round.saturating_sub(round_index));
+			}
+
+			// If this counter is eq to max_offline_rounds,
+			// the collator should be notified as inactive
+			let mut inactive_counter: RoundIndex = RoundIndex::default();
+
+			// Iter rounds to check
+			//
+			// - The collator has AtStake associated and their AwardedPts are zero
+			//
+			// If the previous condition is met in all rounds of rounds_to_check,
+			// the collator is notified as inactive
+			for r in rounds_to_check.clone() {
+				let stake = <AtStake<T>>::get(r, &collator);
+				let pts = <AwardedPts<T>>::get(r, &collator);
+
+				if stake.is_some() && pts.is_zero() {
```

Comment:
`stake.is_some()` will fail when `T::MaxOfflineRounds::get() < T::RewardPaymentDelay::get()`, making it impossible to use the mark-offline feature when this condition is true.

Since both of these are pallet traits, it might be a good candidate for a [integrity_test](https://crates.parity.io/frame_support/pallet_prelude/trait.Hooks.html#method.integrity_test)

Reply:
Why do you think it would fail in that case? I added some asserts to a test case (f492b18300cc17e9e2dd243664ec30b9aa091b3f) that shows that the functionality works fine in the case you are describing.

Reply:
Why do you think it would fail in that case? I added some asserts to a test case (f492b18300cc17e9e2dd243664ec30b9aa091b3f) which shows that the functionality works fine in the case you are describing.

Reply:
I think I was assuming that rewards would have been paid out immediately, clearing `AtStake` because the deferred payout mechanism would execute immediately. Maybe that's wrong or maybe I forgot what I was thinking back in May :)